cmd/common.go

```diff
@@ -602,15 +605,335 @@ func exportCodesignFiles(tool Tool, archivePath, outputDirPath string) error {
+		}
+
+		uploadedCertificatesSerialList = append(uploadedCertificatesSerialList, serialListAsString...)
+	}
+
+	for _, certificateToExport := range certificatesToExport {
+		localCertificatesSerialList = append(localCertificatesSerialList, certificateToExport.Serial)
+	}
+
+	log.Debugf("Uploaded certificates' serial list: \n\t%v", uploadedCertificatesSerialList)
+	log.Debugf("Local certificates' serial list: \n\t%v", localCertificatesSerialList)
+
+	// Search for a new certificate
+	for _, localCertificateSerial := range localCertificatesSerialList {
+		if !sliceutil.IsStringInSlice(localCertificateSerial, uploadedCertificatesSerialList) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// ----------------------------------------------------------------
+// --- Upload methods
+
+func uploadProvisioningProfiles(bitriseClient *bitriseclient.BitriseClient, profilesToUpload []profileutil.ProvisioningProfileInfoModel, outputDirPath string) error {
+	for _, profile := range profilesToUpload {
+		exportFileName := provProfileExportFileName(profile, outputDirPath)
+
+		provProfile, err := os.Open(outputDirPath + "/" + exportFileName)
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if err := provProfile.Close(); err != nil {
+				log.Warnf("Provisioning profile close failed, err: %s", err)
+			}
+
+		}()
+
+		info, err := provProfile.Stat()
+		if err != nil {
+			return err
+		}
+
+		bytes := info.Size()
+		log.Debugf("\n%s size: %d", exportFileName, bytes)
+
+		provProfSlugResponseData, err := bitriseClient.RegisterProvisioningProfile(bytes, profile)
+		if err != nil {
+			return err
+		}
+
+		if err := bitriseClient.UploadProvisioningProfile(provProfSlugResponseData.UploadURL, provProfSlugResponseData.UploadFileName, outputDirPath, exportFileName); err != nil {
+			return err
+		}
+
+		if err := bitriseClient.ConfirmProvisioningProfileUpload(provProfSlugResponseData.Slug, provProfSlugResponseData.UploadFileName); err != nil {
+			return err
+		}
+	}
 
 	return nil
 }
+
+// UploadIdentity ...
+func UploadIdentity(bitriseClient *bitriseclient.BitriseClient, outputDirPath string) error {
+	identities, err := os.Open(outputDirPath + "/" + "Identities.p12")
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := identities.Close(); err != nil {
+			log.Warnf("Identities failed, err: %s", err)
+		}
+
+	}()
+
+	info, err := identities.Stat()
+	if err != nil {
+		return err
+	}
+
+	bytes := info.Size()
+	log.Debugf("\n%s size: %d", "Identities.p12", bytes)
+
+	certificateResponseData, err := bitriseClient.RegisterIdentity(bytes)
+	if err != nil {
+		return err
+	}
+
+	if err := bitriseClient.UploadIdentity(certificateResponseData.UploadURL, certificateResponseData.UploadFileName, outputDirPath, "Identities.p12"); err != nil {
+		return err
+	}
+
+	if err := bitriseClient.ConfirmIdentityUpload(certificateResponseData.Slug, certificateResponseData.UploadFileName); err != nil {
```

Comment:
return bitriseClient.ConfirmIdentityUpload(certificateResponseData.Slug, certificateResponseData.UploadFileName)

Reply:
You can do that but I think it will break the code's continuity. (The previous func call looks the same).
That's why I would not change the return statement.